presto-spi/src/main/java/com/facebook/presto/spi/type/SqlTimeWithTimeZone.java

```diff
@@ -16,30 +16,64 @@
 import com.fasterxml.jackson.annotation.JsonValue;
 
 import java.time.Instant;
+import java.time.LocalTime;
 import java.time.ZoneId;
+import java.time.ZonedDateTime;
 import java.time.format.DateTimeFormatter;
 import java.util.Objects;
 import java.util.TimeZone;
 
 import static com.facebook.presto.spi.type.DateTimeEncoding.unpackMillisUtc;
 import static com.facebook.presto.spi.type.DateTimeEncoding.unpackZoneKey;
 
+/**
+ * This type represent instant of time in some day and timezone, where timezone
+ * is known and day is not.
+ */
 public final class SqlTimeWithTimeZone
 {
     private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("HH:mm:ss.SSS VV");
 
+    /**
+     * As we dont have negative timestamps and max value is midnight in timezone that is ~(UTC -12:00),
+     * just not to care about precision we do use ~(UTC -24:00).
+     *
+     * It's all about protecting type from beeing used with full millisUTC.
+     */
+    private static final long MAX_MILLIS_TIME = 2 * 24 * 60 * 60 * 1000;
+
+    /**
+     * Milliseconds relative to midnight UTC of the given time (millisUtc < MAX_MILLIS_TIME).
```

Comment:
`~(-MAX_MILLIS_TIME)`, do you think we should double sanity checks here and check that as well? I thought that MAX is enough sanity as representing dates before 1970-01-01 in timestamp is rare and I didn't seen this constraint broken in Presto codebase.